tests/testthat/helper-rset.R

```diff
@@ -60,3 +46,23 @@ expect_s3_class_rset <- function(x) {
 expect_s3_class_bare_tibble <- function(x) {
   expect_s3_class(x, c("tbl_df", "tbl", "data.frame"), exact = TRUE)
 }
+
+# ------------------------------------------------------------------------------
+
+# Keep this list up to date with known rset subclasses for testing.
+# Delay assignment because we are creating this directly in the R script
+# and not all of the required helpers might have been sourced yet.
+test_data <- function() {
+  x <- boston_canopy
+  x$idx <- rep(c("a", "b"), length.out = nrow(x))
+  x
+}
+
+delayedAssign("rset_subclasses", {
```

Comment:
We don't need the delayed assign if it is only used in tests and defined in this test helper function. We _do_ need a `delayedAssign()` [in rsample because it's in a regular R file](https://github.com/tidymodels/rsample/blob/main/R/compat-vctrs-helpers.R).